lstchain/scripts/lstchain_dl1_to_dl2.py

```diff
@@ -106,8 +116,20 @@ def main():
     os.makedirs(args.output_dir, exist_ok=True)
     output_file = os.path.join(args.output_dir, os.path.basename(args.input_file).replace('dl1','dl2'))
 
+    if os.path.exists(output_file):
+        raise IOError(output_file + ' exists, exiting.')
+
+    dl1_keys = get_dataset_keys(args.input_file)
+    dl1_keys.remove(dl1_params_lstcam_key)
+    dl1_keys.remove(dl1_images_lstcam_key)
+    if config['source_dependent']:
+        dl1_keys.remove(dl1_params_src_dep_lstcam_key)
+
+    with open_file(args.input_file) as file:
```

Comment:
If you open both files using tables, you can copy datasets:

http://www.pytables.org/usersguide/libref/file_class.html?highlight=copy#tables.File.copy_children

This will be much more efficient than parsing into a table and writing again.
This will also directly preserve metadata, which might be lost with this approach here, right?

Reply:
```
In [7]: with tables.open_file('test.hdf5', 'r') as infile, tables.open_file('out.hdf5', 'w') as out: 
   ...:     g = out.create_group('/', 'data') 
   ...:     infile.copy_node('/data/foo', g) 
```

Reply:
why would metadata be lost?

Reply:
Because `file.root[k][:]` is a numpy structures array, so where should `Table` get the hdf5 attributes from?

Reply:
ok, that sounds a good solution, but maybe we can accept this one for the time being and then propose a general solution to substituted `write_dataframes`, that is used not only here, but also on `lstchain_add_source_dependent_parameters.py` and probably somewhere else, does that sound ok?

Reply:
I gave you the solution above that directly preserves metadata.

Is there a problem with that code? If yes, let's merge and I'll try to address that, if not, let's just use it directly.

Reply:
Does this work?

```
with tables.open_file(args.input_file, 'r') as h5in:
    with tables.open_file(output_file, 'a') as h5out:

        for k in dl1_keys:
            path, name = k.rsplit('/', 1)

            if path not in h5out:
                grouppath, groupname = path.rsplit('/', 1)
                g = h5out.create_group(
                    grouppath, groupname, createparents=True
                )
            else:
                g = h5out.get_node(path)

            h5in.copy_node(k, g, overwrite=True)
```

Reply:
I checked, and this preserves hdf5 attributes in the datasets

Reply:
Thanks for the solution @MaxNoe. Juste needed to add:
```
if not k.startswith('/'):
    k = '/' + k
```
because pytables was complaining